src/integrator_interface.jl

```diff
@@ -584,53 +584,68 @@ function check_error(integrator::DEIntegrator)
-        if integrator.opts.verbose
-            if isdefined(integrator, :EEst)
-                EEst = ", and step error estimate = $(integrator.EEst)"
-            else
-                EEst = ""
+    # Bail out if we take a step with dt less than the minimum value (which may be time dependent)
+    # except if we are successfully taking such a small timestep is to hit a tstop exactly
+    # We also exit if the ODE is unstable (by default this is the same as nonfinite u)
+    # but only consider the ODE unstable if the error is somewhat controlled
+    # (to prevent from bailing out as unstable when we just took way too big a step)
+    if !opts.force_dtmin && opts.adaptive
+        if abs(integrator.dt) <= abs(opts.dtmin) &&
+           (((hasproperty(integrator, :opts) && hasproperty(opts, :tstops)) ?
+             integrator.t + integrator.dt < integrator.tdir * first(opts.tstops) :
+             true) || (hasproperty(integrator, :accept_step) && !integrator.accept_step))
+            if verbose
+                if isdefined(integrator, :EEst)
+                    EEst = ", and step error estimate = $(integrator.EEst)"
+                else
+                    EEst = ""
+                end
+                @warn("dt($(integrator.dt)) <= dtmin($(opts.dtmin)) at t=$(integrator.t)$EEst. Aborting. There is either an error in your model specification or the true solution is unstable.")
+            end
+            return ReturnCode.DtLessThanMin
+        elseif abs(integrator.dt) <= abs(eps(integrator.t)) &&
+                hasproperty(integrator, :accept_step) && !integrator.accept_step
+            if verbose
+                if isdefined(integrator, :EEst)
+                    EEst = ", and step error estimate = $(integrator.EEst)"
+                else
+                    EEst = ""
+                end
+                @warn("dt($(integrator.dt)) <= eps(t)($(integrator.t)) $EEst. Aborting. There is either an error in your model specification or the true solution is unstable (or the true solution can not be represented in the precision of $(eltype(integrator.u))).")
             end
-            @warn("dt($(integrator.dt)) <= dtmin($(integrator.opts.dtmin)) at t=$(integrator.t)$EEst. Aborting. There is either an error in your model specification or the true solution is unstable.")
+            return ReturnCode.Unstable
         end
-        return ReturnCode.DtLessThanMin
-    end
-    if integrator.opts.unstable_check(integrator.dt, integrator.u, integrator.p,
-        integrator.t)
-        bigtol = max(integrator.opts.reltol, integrator.opts.abstol)
-        # only declare instability if the dt is very small
-        # or we have at least one digit of accuracy in the solution
-        if integrator.dt<eps(integrator.t) || isdefined(integrator, :EEst) && integrator.EEst * bigtol < .1
-            if integrator.opts.verbose
+    end
+    bigtol = max(opts.reltol, opts.abstol)
```

Comment:
```suggestion
    bigtol = max(max(opts.reltol), max(opts.abstol))
```

? The array tol case seems to require more

Reply:
good point.